### PR TITLE
Slim down & upgrade Ory charts

### PR DESCRIPTION
**Pull request by ognyvrac**
**Description**
Slim down the Ory charts as much as possible, also, make the Ory charts as close as the opensource ones. The Ory components have been updated to the newest version, this also required some changes around the Go code as it used old versions of the Ory SDK.

As this PR involves a lot of changes, some of them are copy and pasted Helm templates, which are identical to the OS ones - these should not be thoroughly reviewed. I have tried to comprise a list of items that should be the points of interest for the reviewer.

Changes proposed in this pull request:
- Use open source charts for Hydra
- Remove gcloud sqlproxy chart and use OS chart
- The Oathkeeper charts have very minor custom implementations (marked with `# Custom...`). The main focus should be on `chart/ory/charts/oathkeeper/templates/cronjob.yaml` and `chart/ory/charts/oathkeeper/templates/pre-install-job.yaml` (fully custom jobs that handle the rotation and creation of the Oathkeeper JWKS)
- `/chart/ory/values.yaml`
- The `localdb` Helm chart is also used for Ory Hydra
- The Go code changes that came with the new version of the Ory Hydra SDK
- All of the Bash script changes
- `hydra.hydra.automigration` in `/chart/ory/values.yaml` - migration used to be initContainer now is a job
- Workaround for Oathkeeper bug 

**Pull Request status**

- [x] Implementation
- [x] Unit tests
- [ ] Integration tests
- [x] `chart/compass/values.yaml` is updated 
- [x] Mocks are regenerated, using the automated script

### CODE REVIEW

**Comment by nyordanoff** [comment]
What's `chart/ory/charts/hydra-0.36.0.tgz` needed for?

File: chart/ory/charts/oathkeeper/charts/oathkeeper-maester/templates/dashboard-configmap.yaml [hunk 1]
```diff
@@ -1,446 +0,0 @@
-apiVersion: v1
```
[the remark this answers is not in the document]
Comment (ognyvrac) [reply chart/ory/charts/oathkeeper/charts/oathkeeper-maester/templates/dashboard-configmap.yaml:1]: It was not used so it was removed, it also did not show any data. We decided to remove it and not fix

**Comment by ognyvrac** [comment]
> What's `chart/ory/charts/hydra-0.36.0.tgz` needed for?

We are using an umbrella chart and it required that the Hydra chart exists locally. It is created by running `helm dependecy build`.

File: docs/compass/04-01-installation.md [hunk 2]
```diff
@@ -104,23 +113,43 @@ Initiate the Ory installation with the following command:
...
+  instances:
+  - instance: ${GCP_INSTANCE_NAME}
+    port: ${GCP_PORT}
+    project: ${GCP_PROJECT}
+    region: ${GCP_REGION}
+extraFlags:
+  # Uncomment if your database is situated in a private network inside GCP
+  # - private-ip
+```
+> **NOTE:** The overrides above use workload identity to authenticate to Google Cloud Platform(recommended by Google); another possibility is to use the Google Cloud Platform's service account JSON as explained at [gcloud-sqlproxy's installation](https://github.com/rimusz/charts/tree/master/stable/gcloud-sqlproxy#installing-the-chart).
```
Comment (ekaterina-mitova) [inline docs/compass/04-01-installation.md:141]: Space is missing: Google Cloud Platform(recommended ...

File: docs/compass/04-01-installation.md [hunk 3]
```diff
@@ -452,7 +459,7 @@ global:
 #      tlsCrt: ${TLS_CERT}
 #      tlsKey: ${TLS_KEY}
 ```
-Start Database installation:
+Start Database installation, can be skipped if it was installed during [Ory installation with local persistence](#in-cluster-database):
```
Comment (ekaterina-mitova) [inline docs/compass/04-01-installation.md:462]: Start the Database maybe?

**Review by nyordanoff — DISMISSED** [review]

**Review by ekaterina-mitova — APPROVED** [review]
Looks good to me.

**Review by nyordanoff — APPROVED** [review]

**Pull request merged**